dnf5/main.cpp

```diff
@@ -773,29 +774,38 @@ static void print_versions(Context & context) {
 }
 
 static void print_transaction_size_stats(Context & context) {
-    int64_t in_pkgs_size{0};
-    int64_t download_pkgs_size{0};
-    int64_t install_size{0};
-    int64_t remove_size{0};
+    unsigned long long int in_pkgs_size{0};
+    unsigned long long int download_pkgs_size{0};
+    unsigned long long int install_size{0};
+    unsigned long long int remove_size{0};
 
     for (const auto & trans_pkg : context.get_transaction()->get_transaction_packages()) {
         const auto pkg = trans_pkg.get_package();
         if (transaction_item_action_is_inbound(trans_pkg.get_action())) {
             const auto pkg_size = pkg.get_download_size();
             in_pkgs_size += pkg_size;
+            if (in_pkgs_size < pkg_size)
+                return;
             if (!pkg.is_available_locally()) {
                 download_pkgs_size += pkg_size;
+                if (download_pkgs_size < pkg_size)
+                    return;
             }
             install_size += pkg.get_install_size();
+            if (install_size < pkg.get_install_size())
+                return;
         } else if (transaction_item_action_is_outbound(trans_pkg.get_action())) {
             remove_size += pkg.get_install_size();
+            if (remove_size < pkg.get_install_size())
+                return;
         }
     }
 
-    if (in_pkgs_size != 0) {
-        const auto [in_pkgs_size_value, in_pkgs_size_unit] = libdnf5::cli::utils::units::to_size(in_pkgs_size);
+    if (in_pkgs_size != 0 && std::in_range<int64_t>(in_pkgs_size) && std::in_range<int64_t>(download_pkgs_size)) {
```

Comment:
`std::in_range<int64_t>(in_pkgs_size)` - Maybe a bit unnecessary test. The transaction will not exceed 8 exabytes in the near future. And if it does, I suspect there will be more serious problems on other places. :-)